sdk/nodejs/runtime/stack.ts

```diff
@@ -25,16 +27,6 @@ import { isDryRun, isQueryMode, setRootResource } from "./settings";
  */
 export const rootPulumiStackTypeName = "pulumi:pulumi:Stack";
 
-// The stack resource needs to be a a true global, so that if we end up with multiple Pulumi modules loaded
```

Comment:
I think this change is what's causing your sxs issues. 
Current versions of the SDK will expect this global var to be present so we need to keep setting it (obv that's not parallel safe, but that's not expected of old sdk versions, but they should still work for current non-parallel uses)

For supporting sxs & parallelism in new sdks I think the LocalStore needs to be a global var (like stackResource), the issue is that having it as a module var means different sxs modules will get different stores.

Reply:
Don't we need to keep setting `globalThis.sackResource` so older SDK versions continue to use the same one? Otherwise, older sdk and new sdk sxs will be using different instances.

Reply:
I'm not entirely sure - I think we run into a race condition if the sxs run is using different _versions_ rather than just different instances.  How do we know which one is the authoritative stack?  Whichever one is not undefined first?